saleor/graphql/discount/mutations.py

```diff
@@ -382,4 +403,83 @@ def perform_mutation(cls, _root, info, **data):
         )
         cls.remove_catalogues_from_node(sale, data.get("input"))
-        return SaleRemoveCatalogues(sale=sale)
+        return SaleRemoveCatalogues(sale=ChannelContext(node=sale, channel_slug=None))
+
+
+class SaleChannelListingAddInput(graphene.InputObjectType):
+    channel_id = graphene.ID(required=True, description="ID of a channel.")
+    discount_value = PositiveDecimal(description="The value of the discount.")
+
+
+class SaleChannelListingInput(graphene.InputObjectType):
+    add_channels = graphene.List(
+        graphene.NonNull(SaleChannelListingAddInput),
+        description="List of channels to which the sale should be assigned.",
+        required=False,
+    )
+    remove_channels = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description=("List of channels from which the sale should be unassigned."),
+        required=False,
+    )
+
+
+class SaleChannelListingUpdate(BaseChannelListingMutation):
+    sale = graphene.Field(Sale, description="An updated sale instance.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a sale to update.")
+        input = SaleChannelListingInput(
+            required=True,
+            description="Fields required to update sale channel listings.",
+        )
+
+    class Meta:
+        description = "Manage sale's availability in channels."
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = DiscountError
+        error_type_field = "discount_errors"
+
+    @classmethod
+    def add_channels(cls, sale: "SaleModel", add_channels: List[Dict]):
+        for add_channel in add_channels:
+            defaults = {}
+            discount_value = add_channel.get("discount_value")
+            if discount_value:
+                defaults["discount_value"] = discount_value
+            channel = add_channel["channel"]
+            SaleChannelListing.objects.update_or_create(
+                sale=sale,
+                channel=channel,
+                currency=channel.currency_code,
+                defaults=defaults,
+            )
+
+    @classmethod
+    def remove_channels(cls, sale: "SaleModel", remove_channels: List[int]):
+        sale.channel_listing.filter(channel_id__in=remove_channels).delete()
+
+    @classmethod
+    @transaction.atomic()
+    def save(cls, info, sale: "SaleModel", cleaned_input: Dict):
+        cls.add_channels(sale, cleaned_input.get("add_channels", []))
+        cls.remove_channels(sale, cleaned_input.get("remove_channels", []))
+        update_products_discounted_prices_of_discount_task.delay(sale.pk)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, id, input):
+        sale = cls.get_node_or_error(info, id, only_type=Sale, field="id")
+        errors = defaultdict(list)
+
+        clean_channels = cls.clean_channels(
+            info, input, errors, DiscountErrorCode.DUPLICATED_INPUT_ITEM.value
+        )
+
+        if errors:
+
```

Comment:
Maby?
```suggestion
```

Reply:
Agree, in short and simple functions too many additional empty lines doesn't help much with readability.